html/janus.js

```diff
@@ -1624,11 +1624,27 @@ function Janus(gatewayCallbacks) {
 		});
 	}
 
+	function stopAllTracks (stream) {
+		try {
+			// Try a MediaStreamTrack.stop() for each track
+			var tracks = stream.getTracks();
+			for(var mst of tracks) {
+				Janus.log(mst);
+				if(mst)
+					mst.stop();
+			}
+		} catch(e) {
+			// Do nothing if this fails
+		}
+	}
+
 	// WebRTC stuff
 	function streamsDone(handleId, jsep, media, callbacks, stream) {
 		var pluginHandle = pluginHandles[handleId];
 		if(!pluginHandle || !pluginHandle.webrtcStuff) {
 			Janus.warn("Invalid handle");
+			// close all tracks if the given stream has been created internally
+			if (!callbacks.stream) stopAllTracks(stream);
```

Comment:
Code style nits:

1. Comments should start with a capital letter;
2. Please no spaces between `if` and bracket;
3. Please put condition and action on different lines.